File: 1266-minimum-time-visiting-all-points/1266-minimum-time-visiting-all-points.py
class Solution:
    def minTimeToVisitAllPoints(self, points: List[List[int]]) -> int:
        total_time = 0
        for i in range(1, len(points)):
            x1, y1 = points[i-1]
            x2, y2 = points[i]
            dx = abs(x2 - x1)
            dy = abs(y2 - y1)
            total_time += max(dx, dy)
        return total_time

File: 1266-minimum-time-visiting-all-points/1266-minimum-time-visiting-all-points.java
class Solution {
    public int minTimeToVisitAllPoints(int[][] points) {
        int totalTime = 0;
        for (int i = 1; i < points.length; i++) {
            int x1 = points[i - 1][0];
            int y1 = points[i - 1][1];
            int x2 = points[i][0];
            int y2 = points[i][1];
            int dx = Math.abs(x2 - x1);
            int dy = Math.abs(y2 - y1);
            totalTime += Math.max(dx, dy);
        }
        return totalTime;
    }
}